docs/simulators.md

```diff
@@ -41,50 +41,107 @@ suite.
     }
-
-Your simulator directory must also contain a Dockerfile so that hive can build it:
-
-    FROM golang:1-alpine AS builder
-    RUN apk --no-cache add gcc musl-dev linux-headers
-    ADD . /source
-    WORKDIR /source
-    RUN go build -o ./sim .
-
-    # Build the runner container.
-    FROM alpine:latest
-    ADD . /
-    COPY --from=builder /source/sim /
-    ENTRYPOINT ["./sim"]
+	// add a plain test (does not run a client) (comment to remove)
+	suite.Add(hivesim.TestSpec{
+		Name:        "the-test",
+		Description: "This is the example test case.",
+		Run: runMyTest,
+	})
+	// add a client test (starts the client)
+    suite.Add(hivesim.ClientTestSpec{
+        Name:        "the-test",
+        Description: "This is the example test case.",
+        Files: map[string]string{"/genesis.json": "genesis.json"},
+        Run: runMyClientTest,
+    })
+	// runs the tests (exist on failures) (RunSuit does not exit) 
+    hivesim.MustRunSuite(hivesim.New(), suite)
+}
+
+
+// template for the example test
+func runMyTest(t *hivesim.T) {
+	// write your test code here
+}
+
+// template for the example client test
+func runMyClientTest(t *hivesim.T, c *hivesim.Client) {
+	// write your test code here
+}
+```
+
+The templates of the funcions are to be filled with the actuall
+code. To do that you need to program the test routines using hivesim
+package. The [documentation of hivesim][hivesimref] provides a lot of
+useful information about the functions you can use.
+
+### Setup a simulation container
+
+The simulation runs in a docker container. For that we need to setup a
+correct environment. As can be seen in the client test `Files:` part,
+it requires a `genesis.json` file that specifies the genesis state of
+the client. On example of `genesis.json` can be found in
+`simulators/devp2p/eth/init/genesis.json`. Copy it to your current
+`my-simulation` directory (as `cp
+../../../simulators/devp2p/eth/init/genesis.json .`.
+
+Your simulator directory must also contain a `Dockerfile` so that hive
+can build the container:
+
+```
+FROM golang:1-alpine AS builder
+RUN apk --no-cache add gcc musl-dev linux-headers
+ADD . /source
+WORKDIR /source
+RUN go build -o ./sim .
+
+# Build the runner container.
+FROM alpine:latest
+ADD . /
+COPY --from=builder /source/sim /
+ENTRYPOINT ["./sim"]
+```
+
+### Running the simulation
+
+Finaly, go back to the root of the repository (`cd ../../..`) and [launch the simulation]:
+
+	./hive --sim my-simulation --client go-ethereum,openethereum
+
+[launch the simulation]: [./overview.md#running-hive]
+
+### Checking the results
+
+The test results are going to be written to the standard output and
+you can also review the results [on your server in case you run the
+`hiveview`][hiveview]. If you set everythin as described, your test
+should pass and you have your testing environment set up correctly.
+
+[hiveview]: ./commandline.md#viewing-simulation-results-hiveview
```

Comment:
I think a lot of this content you added here overlaps with the other documents.

Reply:
Well, you might be right, but I've read the whole documentation in this folder and many thinks I've added there were not mentioned yet. I've tried to formulate some of the overlapping information more clearly. Also, as I've got through the initial setup following the documentation, and the tests failed. So, I've completed the code including a dummy functions for instance, so that that allow the tests to pass. Finally, I've changed the name of the files from `my-test` to `my-simulation` as @protolambda suggested that the name `test` is reserved for other things.